cli/server_report_command.go

```diff
@@ -1540,3 +1548,118 @@ func (c *SrvReportCmd) reqFilter() server.EventFilterOptions {
 		Tags:    c.tags,
 	}
 }
+
+type jsDowngradeAsset struct {
+	Streams   []jsDowngradeStreamAsset   `json:"streams,omitempty"`
+	Consumers []jsDowngradeConsumerAsset `json:"consumers,omitempty"`
+}
+
+type jsDowngradeStreamAsset struct {
+	Account  string `json:"account"`
+	Name     string `json:"name"`
+	APILevel uint   `json:"api_level"`
+}
+
+type jsDowngradeConsumerAsset struct {
+	Account  string `json:"account"`
+	Name     string `json:"name"`
+	Stream   string `json:"stream"`
+	APILevel uint   `json:"api_level"`
+}
+
+func (c *SrvReportCmd) downgradeCheckAction(_ *fisk.ParseContext) error {
+	nc, err := newNatsConn("", natsOpts()...)
+	if err != nil {
+		return fmt.Errorf("connection setup failed: %v", err)
+	}
+	defer nc.Close()
+
+	if !c.json {
+		fmt.Println("Obtaining JetStream Asset information")
+	}
+
+	sys := sysclient.New(nc)
+	accounts, err := sys.CollectClusterAccounts(opts().Timeout)
```

Comment:
timeout doesn't seem to be handled right

```
[rip@s1-lon]% bin/nats srv report downgrade 0 --context system.lon --all --timeout 1ms
Obtaining JetStream Asset information
All assets are compatible with the specified API level.
```

when in fact there are - fetching just timed out, some timeout error isnt propagating up